lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -260,25 +264,53 @@ xercesc::DOMNode* ParseXML::get_node(
                                       tag_name + " does not have the attribute " + att_name + "\n");
                         }
                     }
-                    // LIST element TODO refactor
-                    else if (&index == nullptr)
+                    // LIST element
+                    else if (index != nullptr)
                     {
-                        int32_t realIndex = index;
-                        // Transform index to real index
-                        if (index < 0)
+                        // Check value of index
+                        std::string in = *index;
```

Comment:
Passing a reference will avoid the necessity of accessing the content of the pointer, so this instruction can be removed.